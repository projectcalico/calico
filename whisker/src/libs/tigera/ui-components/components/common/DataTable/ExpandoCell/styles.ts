export const tableExpandoCellStyles = {
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
};
