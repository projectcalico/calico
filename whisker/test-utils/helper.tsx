import React, { ReactElement } from 'react';
import { render, renderHook, RenderOptions } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { theme } from '../src/theme';
import { MemoryRouter } from 'react-router';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
        },
    },
});

const queryClientProvider = ({ children }: any) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

export const renderHookWithQueryClient = (hook: () => unknown) =>
    renderHook(hook, { wrapper: queryClientProvider });

const Providers = ({ children }: { children: React.ReactNode }) => {
    return <ChakraProvider theme={theme}>{children}</ChakraProvider>;
};

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Providers, ...options });

export const renderWithRouter = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'> & {
        routes: string[];
    },
) =>
    customRender(
        <MemoryRouter initialEntries={options?.routes}>{ui}</MemoryRouter>,
        options,
    );

export * from '@testing-library/react';

export { customRender as render };
