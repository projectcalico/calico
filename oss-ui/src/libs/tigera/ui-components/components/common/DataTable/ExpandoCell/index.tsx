import * as React from 'react';
import { Box } from '@chakra-ui/react';
import { ChevronRightIcon, ChevronDownIcon } from '@chakra-ui/icons';
import type {
    SystemStyleObject,
    HTMLChakraProps,
    IconProps,
} from '@chakra-ui/react';
import { tableExpandoCellStyles } from './styles';

interface ExpandoCellProps extends HTMLChakraProps<'div'> {
    sx?: SystemStyleObject;
    isExpanded: boolean;
    value: string;
    iconProps?: IconProps;
}

const ExpandoCell = React.forwardRef(
    ({ isExpanded, value, sx, iconProps, ...rest }: ExpandoCellProps, ref) => (
        <Box
            ref={ref as any}
            data-testid={'expando-cell'}
            sx={{ ...tableExpandoCellStyles, ...sx }}
            {...rest}
        >
            {isExpanded ? (
                <ChevronDownIcon w={5} h={5} mr={1} {...iconProps} />
            ) : (
                <ChevronRightIcon w={5} h={5} mr={1} {...iconProps} />
            )}
            {value}
        </Box>
    ),
);

ExpandoCell.displayName = 'ExpandoCell';

export default ExpandoCell;
