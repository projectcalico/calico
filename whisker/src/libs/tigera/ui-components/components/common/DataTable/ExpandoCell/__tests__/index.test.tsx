import { render } from '@/test-utils/helper';
import ExpandoCell from '../index';

describe('<ExpandoCell/>', () => {
    it('it renders', () => {
        const { asFragment } = render(
            <ExpandoCell isExpanded={false} value='Some Cell Value' />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('it renders inactive', () => {
        const { asFragment } = render(
            <ExpandoCell isExpanded={true} value='Some Cell Value' />,
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
