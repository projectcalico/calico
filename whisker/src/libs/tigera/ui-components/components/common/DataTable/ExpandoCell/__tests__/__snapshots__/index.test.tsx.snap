// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandoCell/> it renders 1`] = `
<DocumentFragment>
  <div
    class="css-1h2ruwl"
    data-testid="expando-cell"
  >
    <svg
      class="chakra-icon css-k2ty79"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        fill="currentColor"
      />
    </svg>
    Some Cell Value
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<ExpandoCell/> it renders inactive 1`] = `
<DocumentFragment>
  <div
    class="css-1h2ruwl"
    data-testid="expando-cell"
  >
    <svg
      class="chakra-icon css-k2ty79"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        fill="currentColor"
      />
    </svg>
    Some Cell Value
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
